kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/FilterFactoryContext.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import java.util.concurrent.ScheduledExecutorService;
+
+import io.kroxylicious.proxy.plugin.UnknownPluginInstanceException;
+import io.kroxylicious.proxy.plugin.UnknownPluginTypeException;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Construction context for Filters. Used to pass the filter configuration and environmental resources
+ * to the FilterFactory when it is creating a new instance of the Filter. see {@link FilterFactory#createFilter(FilterFactoryContext, Object)}
+ */
+public interface FilterFactoryContext {
+
+    /**
+     * The event loop that this Filter's channel is assigned to.
+     * Null if the factory is not bound to a channel yet.
+     * Should be safe
+     * to mutate Filter members from this executor.
+     * @return executor, or null
+     */
+    ScheduledExecutorService eventLoop();
+
+    /**
+     * Gets a plugin instance for the given plugin type and name
```

Comment:
Does this imply that plug-in instances are cacheable?

Reply:
If you mean: "Can the returned instance be cache by the caller of this method?" then I think that depends on the contract of the plugin interface.

If you mean "Can this returned a cached instance to the caller?" then it boils down to the contract that the `ServiceLoader.Provider.get()` offers. That will _usually_ invoke the nullary constructor of the service implementation. But it also supports static methods, which could return the same instance.